R/scope.R

```diff
@@ -329,13 +329,13 @@ renv_scope_trace <- function(what, tracer, envir = parent.frame()) {
 }
 
 
-renv_scope_binding <- function(envir, symbol, replacement, frame = parent.frame()) {
-  if (exists(symbol, envir, inherits = FALSE)) {
-    old <- renv_binding_replace(symbol, replacement, envir)
-    defer(renv_binding_replace(symbol, old, envir), envir = frame)
+renv_scope_binding <- function(symbol, replacement, frame = renv_envir_self(), envir = parent.frame()) {
```

Comment:
Yeah, this looks good. I still have a note to go through and systematically change `frame`/`envir` to `scope` in all the scoped functions.